### PR TITLE
Migrate GandiProvider to new LiveDns api

### PR DESCRIPTION
**Pull request by bhubert**
Gandi API has changed and migrate to new endpoints

### CODE REVIEW

**Comment by shibayan** [comment]
This appears to be a disruptive change to users using v4, has it been tested to be OK?

**Comment by bhubert** [comment]
> This appears to be a disruptive change to users using v4, has it been tested to be OK?

i tested my new version since i'm generating certificate for my company with my new fork and it's working great !

But for users on v4 i can't test it since my company is on v5. I'm just sure that the current version is not working on gandi v5 (the security header itself has changed from X-API-KEY to classic Authorization for example).

If you prefer i can add a option like UseLiveDNS in the Gandi Provider which will be at false by default and handle the two apis (v4 and v5) in the provider

**Comment by bhubert** [comment]
@shibayan I reworked the code to make it work with legacy api by default and allow new gandi users to use the new api

**Comment by shibayan** [comment]
Do not change the existing Gandi DNS implementation, but add the new implementation as GandiV5

**Comment by bhubert** [comment]
> Do not change the existing Gandi DNS implementation, but add the new implementation as GandiV5

The old one was already using "V5" in its path, its not really the version that changed, its more gandi that created new api endpoints (that started with v5).

So i changed as you asked by adding a new Provider but i called it GandiLiveDnsProvider and not GandiV5.

Hope it will suit you, thanks for all your fast feedbacks !

File: KeyVault.Acmebot/Providers/GandiLiveDnsProvider.cs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        return _client.AddRecordAsync(zone.Name, relativeRecordName, values);
+    }
+
+    public Task DeleteTxtRecordAsync(DnsZone zone, string relativeRecordName)
+    {
+        return _client.DeleteRecordAsync(zone.Name, relativeRecordName);
+    }
+
+    private class GandiClient
```
Comment (shibayan) [inline KeyVault.Acmebot/Providers/GandiLiveDnsProvider.cs]: Please rename `GandiClient` => `GandiLiveDnsClient`

Comment (bhubert) [reply KeyVault.Acmebot/Providers/GandiLiveDnsProvider.cs]: Done

**Review by shibayan — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**